app/javascript/src/modules/media_thumbnail_builder.js

```diff
@@ -9,7 +9,7 @@ export default function(dataset, index) {
     }
 
     let thumbnailIcon = '';
-    if (dataset.thumbnailUrl !== '') {
+    if (dataset.thumbnailUrl && dataset.thumbnailUrl !== '') {
```

Comment:
The bug here was that the `img@src` attr below was being set to `undefined` because the check above only looks at whether `thumbnailUrl` is an empty string, not if it's defined.

If folks prefer that this bug should be fixed upstream of this code, holler at me.

Reply:
Yes, I'm wondering why it wasn't defined. I think we should track that down too.

Reply:
@jcoyne Will do

Reply:
@jcoyne When the file has no thumbnail (when `file.thumbnail` is falsey), `file_thumb` is set to `nil` here: https://github.com/sul-dlss/sul-embed/blob/db08304bd05e518bfe11bd0495b6108b13e369ef/app/components/embed/media_tag_component.rb#L37

Reply:
okay, so `data: { thumbnail_url: nil }` ends up not generating any attribute, thus we need to do the check on `dataset.thumbnailUrl`

Reply:
How about we switch it to `data: { thumbnail_url: @thumbnail.presence }` and then we can drop the empty string check in the javascript?

Reply:
Sounds good, as I don't think it's ever finding the empty string now.